src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Emitter.cs

```diff
@@ -2,41 +2,40 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
+using System.Collections.Generic;
 using System.Diagnostics;
 using System.Text.RegularExpressions;
 using Microsoft.CodeAnalysis;
-using SourceGenerators;
+using DotnetRuntime.SourceGenerators;
 
 namespace Microsoft.Extensions.Configuration.Binder.SourceGeneration
 {
     public sealed partial class ConfigurationBindingGenerator : IIncrementalGenerator
     {
         private sealed partial class Emitter
         {
-            private readonly SourceProductionContext _context;
-            private readonly SourceGenerationSpec _sourceGenSpec;
+            private readonly GenerationSpec _genSpec;
+            private Dictionary<TypeRef, TypeSpec> _typeIndex = null!;
 
-            private bool _emitBlankLineBeforeNextStatement;
+            private bool _emitBlankLineBeforeNextBlock;
             private bool _useFullyQualifiedNames;
             private int _valueSuffixIndex;
 
             private static readonly Regex s_arrayBracketsRegex = new(Regex.Escape("[]"));
 
             private readonly SourceWriter _writer = new();
 
-            public Emitter(SourceProductionContext context, SourceGenerationSpec sourceGenSpec)
-            {
-                _context = context;
-                _sourceGenSpec = sourceGenSpec;
-            }
+            public Emitter(GenerationSpec sourceGenSpec) => _genSpec = sourceGenSpec;
 
-            public void Emit()
+            public void Emit(SourceProductionContext context)
             {
                 if (!ShouldEmitBinders())
                 {
                     return;
                 }
 
+                _typeIndex = _genSpec.GetTypeIndex();
```

Comment:
Consider doing this on the constructor to avoid the `null!` suppression in line 18.